### PR TITLE
Closes #7065: 3.18 - Filter changing output

### LINKED ISSUE #7065: 3.18 - Filter to change the output method
**User Story**
As a user, I’d like to be able to print the Google Fonts CSS as an internal style

**Acceptance Criteria**
- By default, the feature store CSS in the file, it'd be possible to change it with the filter and print it as internal CSS
- The feature will keep the original structure as `data-wpr-hosted-gf-parameters` argument

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Fixes #7065 
It allows the user to use the new added filter `rocket_internal_fonts_styling` in order to change the way we are including the local hosted fonts. 

If `rocket_host_fonts_locally_inline_css ` returns true => We get the css within a `<style></style>` tag
if `rocket_host_fonts_locally_inline_css ` returns false => We get the css within a `<link rel="stylesheet" [...]>` tag

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality).
- [x] Sub-task of #7063 

## Detailed scenario
Just set the filter mentioned in the description above to `true` and see the change on the front-end.

### Documentation

Refer to the description, pretty simple. 

### Risks

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style
- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

### CODE REVIEW

File: inc/Engine/Media/Fonts/Frontend/Controller.php [hunk 1]
```diff
@@ -104,6 +123,15 @@ protected function get_optimized_markup( string $hash, string $original_url ): s
 
 		$gf_parameters = wp_parse_url( $original_url, PHP_URL_QUERY );
 
+		$internal_styling = wpm_apply_filters_typed( 'boolean', 'rocket_internal_fonts_styling', false );
+		if ( $internal_styling ) {
```
Comment (remyperona) [inline inc/Engine/Media/Fonts/Frontend/Controller.php]:
```suggestion
		$inline_fonts_css = wpm_apply_filters_typed( 'boolean', 'rocket_host_fonts_locally_inline_css', false );
		if ( $inline_fonts_css ) {
```

Need the docblock too

File: inc/Engine/Media/Fonts/Frontend/Controller.php [hunk 2]
```diff
@@ -104,6 +123,15 @@ protected function get_optimized_markup( string $hash, string $original_url ): s
...
+			$raw_path       = $this->base_path . $path . '.css';
+			$internal_style = $this->set_font_internal_style( $gf_parameters, $raw_path );
+			if ( $internal_style ) {
+				return $internal_style;
+			}
```
Comment (remyperona) [inline inc/Engine/Media/Fonts/Frontend/Controller.php]:
```suggestion
			$inline_css = $this->get_font_inline_css( $gf_parameters, $raw_path );
			if ( $inline_css ) {
				return $inline_css;
			}
```

File: inc/Engine/Media/Fonts/Frontend/Controller.php [hunk 3]
```diff
@@ -125,4 +153,28 @@ public function disable_google_fonts_preload( $disable ): bool {
 
 		return true;
 	}
+
+	/**
+	 * Sets the font internal style.
```
Comment (remyperona) [inline inc/Engine/Media/Fonts/Frontend/Controller.php]:
```suggestion
	 * Gets the font inline css.
```

File: inc/Engine/Media/Fonts/Frontend/Controller.php [hunk 4]
```diff
@@ -125,4 +153,28 @@ public function disable_google_fonts_preload( $disable ): bool {
...
+	 *
+	 * @param string $gf_parameters Google Fonts parameters.
+	 * @param string $css_path CSS file path.
+	 *
+	 * @return string|bool
+	 */
+	private function set_font_internal_style( string $gf_parameters, string $css_path ) {
```
Comment (remyperona) [inline inc/Engine/Media/Fonts/Frontend/Controller.php]:
```suggestion
	private function get_font_inline_css( string $gf_parameters, string $css_path ) {
```

**Review by remyperona — CHANGES_REQUESTED** [review]

File: inc/Engine/Media/Fonts/ServiceProvider.php [hunk 5]
```diff
@@ -46,6 +49,15 @@ public function provides( string $id ): bool {
 	 */
 	public function register(): void {
 		$this->getContainer()->add( 'media_fonts_settings', Settings::class );
+		$this->getContainer()->add( 'wp_direct_filesystem', WP_Filesystem_Direct::class )
+			->addArgument( [] );
+		$this->getContainer()->add( 'media_fonts_filesystem', Filesystem::class )
+			->addArguments(
+				[
+					wpm_apply_filters_typed( 'string', 'rocket_host_font_cache_root', 'fonts/' . get_current_blog_id() ),
```
Comment (remyperona) [inline inc/Engine/Media/Fonts/ServiceProvider.php]: Is this path needed to be filterable from the AC?

**Review by Khadreal — APPROVED** [review]

**Pull request merged**